src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -207,12 +207,6 @@ else if (shouldBeOnSameLine(option, details)) {
         else if (shouldBeAloneOnLine(option, details)) {
             violation = MSG_KEY_LINE_ALONE;
         }
-        else if (shouldStartLine) {
```

Comment:
Revert all changes that are not in `SingleSpaceSeparatorCheckTest` or it's inputs.